cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/CapacityConfiguration.java

```diff
@@ -0,0 +1,257 @@
+ *
+ * {
+ *   "brokerCapacities": [
+ *     {
+ *       "brokerId": "0",
+ *       "capacity": {
+ *         "DISK": {
+ *           "/var/lib/kafka0/kafka-log0": "100000",
+ *           "/var/lib/kafka1/kafka-log0": "200000"
+ *         },
+ *         "CPU": { "num.cores": "2.345" },
+ *         "NW_IN": "10000",
+ *         "NW_OUT": "40000"
+ *       },
+ *       "doc": "Capacity for Broker 0"
+ *     },
+ *     {
+ *       "brokerId": "1",
+ *       "capacity": {
+ *         "DISK": {
+ *           "/var/lib/kafka0/kafka-log1": "100000",
+ *           "/var/lib/kafka1/kafka-log1": "200000"
+ *         },
+ *         "CPU": { "num.cores": "4" },
+ *         "NW_IN": "60000",
+ *         "NW_OUT": "20000"
+ *       },
+ *       "doc": "Capacity for Broker 1"
+ *     },
+ *     {
+ *       "brokerId": "2",
+ *       "capacity": {
+ *         "DISK": {
+ *           "/var/lib/kafka0/kafka-log2": "100000",
+ *           "/var/lib/kafka1/kafka-log2": "200000"
+ *         },
+ *         "CPU": { "num.cores": "4" },
+ *         "NW_IN": "60000",
+ *         "NW_OUT": "20000"
+ *       },
+ *       "doc": "Capacity for Broker 2"
+ *     }
+ *   ]
+ * }
+ */
+public class CapacityConfiguration {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(CapacityConfiguration.class.getName());
+
+    private final TreeMap<Integer, CapacityEntry> capacityEntries;
+
+    /**
+     * Constructor
+     *
+     * @param reconciliation        Reconciliation marker.
+     * @param spec                  Spec of the Kafka custom resource.
+     * @param kafkaBrokerNodes      List of the broker nodes which are part of the Kafka cluster.
+     * @param kafkaStorage          A map with storage configuration used by the Kafka cluster and its node pools.
+     * @param kafkaBrokerResources  A map with resource configuration used by the Kafka cluster and its broker pools.
+     */
+    public CapacityConfiguration(
+            Reconciliation reconciliation,
+            KafkaSpec spec,
+            Set<NodeRef> kafkaBrokerNodes,
+            Map<String, Storage> kafkaStorage,
+            Map<String, ResourceRequirements> kafkaBrokerResources
+    ) {
+        this.capacityEntries = generateCapacityEntries(reconciliation, spec.getCruiseControl(), kafkaBrokerNodes,
+                kafkaStorage, kafkaBrokerResources);
+    }
+
+    private static Map<Integer, BrokerCapacityOverride> processBrokerCapacityOverrides(Reconciliation reconciliation,
+                                                                                       Set<NodeRef> kafkaBrokerNodes,
+                                                                                       BrokerCapacity brokerCapacity) {
+        Map<Integer, BrokerCapacityOverride> overrideMap = new HashMap<>();
+
+        if (brokerCapacity != null && brokerCapacity.getOverrides() != null && !brokerCapacity.getOverrides().isEmpty()) {
```

Comment:
I think the last check on empty can be dropped because the for loop will be just skipped with an empty collection.

Reply:
Fixed.